craft_cli/pytest_plugin.py

```diff
@@ -158,6 +158,19 @@ def assert_verbose(
         """
         return self._check(expected_text, "verbose", regex)
 
+    def assert_warning(
+        self,
+        expected_text: str,
+        regex: bool = False,  # noqa: FBT001
```

Comment:
Since this is a brand new method we can (and imo should) make `regex` a keyword-only argument

Reply:
Agreed. We can make the change for the rest of the existing methods in the next major release.